singer/src/main/java/com/pinterest/singer/processor/DefaultLogStreamProcessor.java

```diff
@@ -549,7 +550,18 @@ private void writeLogMessages(List<LogMessageAndPosition> logMessagesRead)
     }
     List<LogMessage> logMessagesToWrite = Lists.newArrayListWithExpectedSize(numMessages);
     for (LogMessageAndPosition logMessageRead : logMessagesRead) {
-      logMessagesToWrite.add(logMessageRead.getLogMessage());
+      LogMessage logMessage = logMessageRead.getLogMessage();
+      logMessagesToWrite.add(logMessage);
+      OpenTsdbMetricConverter.gauge(
+              SingerMetrics.PROCESSOR_MESSAGE_KEY_SIZE_BYTES,
+              logMessage.getKey().length,
```

Comment:
This will throw an NPE when key is null

Reply:
Good point. I think we should still send a metric if key is null so we can correlated values for key/value pairs. Would it make sense to send a fixed number like `-1` when key is null?

Reply:
If the key is null the bytes should be 0. -1 will provide an incorrect signal.